README.md

```diff
@@ -73,6 +73,8 @@ When building Swift, keep the following in mind:
 
 To checkout a specific version of Swift, use the `update-checkout` script. For example: `./swift/utils/update-checkout --clone --tag swift-5.0.1-RELEASE`.
 
+_note: `./swift/utils/update-checkout` may fail and exit successfully if the tag doesn't exist. e.g. llvm doesn't have the tag swift-5.0.1-RELEASE. Fix by checking out the swift-5.0-branch_
```

Comment:
hmm, I see `swift-5.0.1-RELEASE`. I wonder if you need to run `git fetch --tags`? I don't know why this would be needed.

```
% git tag --list | rg swift-5.+REL
swift-5.0-RELEASE
swift-5.0.1-RELEASE
swift-5.0.2-RELEASE
swift-5.0.3-RELEASE
swift-5.1-RELEASE
swift-5.1.1-RELEASE
```

Reply:
Hey @kastiglione - this happens only for `apple/llvm-project`, where these tags aren't loaded by the script ( see below repro steps )

```
$ git remote -v
origin  https://github.com/apple/llvm-project.git (fetch)
origin  https://github.com/apple/llvm-project.git (push)
$ git tag --list | grep swift
```

I'm also fairly certian that the tags don't exist, as after running `$ git fetch --tags` I still don't see the tags.

For me, this consistency reproduce on fresh / ephemeral env:

```
set -e

# Install required deps
/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
brew install cmake ninja


# Checkout swift at master ( consider doing shallow )
mkdir swift-source
cd swift-source
git clone https://github.com/apple/swift.git

./swift/utils/update-checkout --clone --tag swift-5.0.1-RELEASE

# Per the suggested steps ( skip build )
./swift/utils/build-script --release --skip-build --llvm-targets-to-build X86

# CMake Error at stdlib/public/SwiftShims/CMakeLists.txt:104 (message):
#   Clang headers were not found in any of the following locations:
#   /Users/pinployee/swift-repo/build/Ninja-ReleaseAssert/llvm-macosx-x86_64/./lib/clang/7.0.0;/Users/pinployee/swift-repo/build/Ninja-ReleaseAssert/llvm-macosx-x86_64/./lib/clang/7.0.0;/Users/pinployee/swift-repo/build/Ninja-ReleaseAssert/llvm-macosx-x86_64/Release/lib/clang/7.0.0
# Call Stack (most recent call first):
#   stdlib/public/SwiftShims/CMakeLists.txt:109 (find_llvm_clang_headers)
#
#
# -- Configuring incomplete, errors occurred!

# The checkout script does not checkout llvm on 5.0.1 604054e43066643ee7582f3f0622a01cdae916f1 (HEAD, tag: swift-5.0.1-RELEASE)
# So you need to do this
cd llvm && git checkout origin/swift/swift-5.0-branch

....
```

Reply:
Edit, my original assertion around it failing silently is wrong. It does actually give errors and exit non-zero:

```
======UPDATE FAILURES======
/Users/pinployee/swift-repo/swift failed (ret=1): ['git', 'checkout', 'swift-5.0.1-RELEASE']
error: pathspec 'swift-5.0.1-RELEASE' did not match any file(s) known to git

update-checkout failed, fix errors and try again
$ echo $?
1
````

Reply:
Synced up offline, I needed to run the right version of `update-checkout` - updated the docs to add a note of this.